sycl/test/check_device_code/matrix/matrix-nvptx-single-bit-test.cpp

```diff
@@ -0,0 +1,76 @@
+// REQUIRES: cuda
+
+// RUN: %clangxx -fsycl-device-only -fsycl-targets=nvptx64-nvidia-cuda -Xsycl-target-backend --cuda-gpu-arch=sm_80 -DSYCL_EXT_ONEAPI_MATRIX=3 -S -Xclang -emit-llvm %s -o -| FileCheck %s
+
+#include <CL/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental::matrix;
+
+// M, N, (K * 32) define the sizes of dimensions of the three matrix types (a,
+// b, accumulator) used per subgroup operation.
+constexpr int M = 8; // number of rows of accumulator,
+                     // number of cols of b.
+constexpr int N = 8; // number of cols of accumulator,
+                     // number of rows of a.
+constexpr int K = 4; // number of cols of a/number of rows of b divided by 32
```

Comment:
>>> number of cols of a/number of rows of b divided by 32
should be:
number of bits in cols of a/number of bits in rows of B divided by 32.
If this is true, do we need to add the "divided by 32" in the code example.
I meant before to add the "multiplies by 32" in the implementation code to explain that this is how we get number of bits that exist in the intrinsics. But at the user level code, is this needed?

Reply:
Here K=4 is not the number of cols in a subgroup matrix: you have to multiply by 32, since K gives a dimension of the arrays A/B which hold the single-bit matrix elements in uint32_t storage type.  There are 32 single-bit matrix elements per uint32_t storage type.

I tried to describe the purpose of these bitwise matrix multiplications [here](https://github.com/intel/llvm/blob/ecb8e6bc1f75d7a3599c8b894d9b02a163b2842a/sycl/doc/extensions/Matrix/dpcpp-joint-matrix.asciidoc#motivation-for-bmad) without going into too much detail.  I added references for full details on the origins of the single-bit models and how they use the bitwise matrix multiplications within them. I do not find references to the usage of such "bitwise matrix multiplications" outside of such models (although of course this means there could be/will be in the future), but I think that
this functionality was introduced specifically with such a use case in mind.

It is important for such users to understand that each bit is considered an element of the matrix by `joint_matrix_bmad` (the matrix element is "quantized" to a single bit), which is why in the original implementation I set K = 128.  However as you pointed out this leads to lots of factors of 32 because we have to divide by 32 to get the number of uint32_t array elements that are used to store the matrix.  
In the current implementation it is nice that these factors are gone, but there should still be proper documentation describing the relationship between "K" and the actual number of (single-bit) matrix elements. Since this is experimental I think it is normal to expect that once people start using this there could be feedback suggesting small changes to the interface: I'm not sure whether the interface I originally set up that led to the factors of 32 or the one you suggested is preferable for the users, but I imagined that at this experimental stage it can (and I imagine most likely will!) be changed in some way in the future anyway.

I could add back the naming scheme A -> A_Packed, B-> B_Packed that I originally used when I switched to K=128 -> K=4 to make is clearer that I am calling "a" the matrix and "A_Packed" a packed array representation of the matrix?
Then I could also add some more detailed description in both the tests and the implementation?  I did not want to go into too much detail in tests/implementation because I thought that the proper place for such descriptions would be in the documentation of the extension? This is why I kept things concise here and did not mention details in the implementation.